src/app/functions/server/getRelevantGuidedSteps.ts

```diff
@@ -206,8 +218,11 @@ function hasCompleted(data: InputData, stepId: StepLink["id"]): boolean {
     dataClass: (typeof BreachDataTypes)[keyof typeof BreachDataTypes]
   ): boolean {
     return !data.subscriberBreaches.some((breach) => {
+      const effectedDataClasses = breach.dataClassesEffected.map(
+        (dataClassEffected) => Object.keys(dataClassEffected).join("")
```

Comment:
I think this results in matching on substrings? Whereas `.includes` should also work fine on array elements, and additionally not accidentally match on other strings that happen to start with the same characters.

```suggestion
        (dataClassEffected) => Object.keys(dataClassEffected)
```

Reply:
The actual matching happens in `L225` — `dataClassEffected` will be an object with only one key. Here I’m only getting the key/name of the affected data class. The matching type is:
```
export type DataClassEffected = {
  [dataType: string]: number | string[];
};
```

It might be clearer if I write:
```suggestion
        (dataClassEffected) => Object.keys(dataClassEffected)[0]
```
?

Reply:
Updated that line in [daee6c9](https://github.com/mozilla/blurts-server/pull/3493/commits/daee6c99c6e396e7416c20afbaa1c872dbeb82e7).